openprescribing/pipeline/middleware.py

```diff
@@ -2,17 +2,32 @@
 from pipeline.runner import in_progress
 
 
+IMPORT_WARNING_TAG = 'import_warning'
+
+
 def import_warning_middleware(get_response):
     def middleware(request):
-        if in_progress() and not request.path.startswith('/api/'):
-            # If we add a message to an API request, then depending on whether
-            # the API requests get made in parallel, Django can end up updating
-            # the cookie multiple times, causing the message to be displayed
-            # multiple times.
+        if _should_add_import_warning(request):
             msg = 'We are currently importing the latest prescribing data.  You may see incomplete data across the site.  Please check back in a couple of hours!'
-            messages.add_message(request, messages.WARNING, msg)
+            messages.add_message(request, messages.WARNING, msg, extra_tags=IMPORT_WARNING_TAG)
 
         response = get_response(request)
 
         return response
     return middleware
+
+
+def _should_add_import_warning(request):
+    if not in_progress():
+        return False
+    # If we add a message to an API request, then depending on whether the API
+    # requests get made in parallel, Django can end up updating the cookie
+    # multiple times, causing the message to be displayed multiple times.
```

Comment:
As we have the more robust check below, we could change this comment to something like: "no need to set the warning on API requests".